### PR TITLE
update nuget and sdk rtm-2241

### PR DESCRIPTION
**Pull request by rohit21agrawal**
CC: @piotrp @rrelyea

### CODE REVIEW

**Comment by rohit21agrawal** [comment]
@dotnet-bot test this please

**Comment by rohit21agrawal** [comment]
@dotnet-bot test this please.

**Comment by rohit21agrawal** [comment]
@dotnet-bot test this please

**Comment by livarcocc** [comment]
You may be hitting the issues that I fixed for rc3, which is restore failing for projects that don't have a packagereference.

I can be seen below, the project that is failing to restore has no un-conditional package reference. Though to be honest, I thought that the implicit package reference would be sufficient for this.

Anyways, this is just a theory. We can try it out tomorrow.

```
<ItemGroup Condition=" '$(TargetFramework)' == 'netcoreapp1.0' ">
    <PackageReference Include="Microsoft.NETCore.App">
      <Version>1.0.3</Version>
    </PackageReference>
  </ItemGroup>

  <ItemGroup Condition=" '$(TargetFramework)' == 'net451' ">
    <Reference Include="System" />
    <Reference Include="Microsoft.CSharp" />
  </ItemGroup>
```

**Comment by rohit21agrawal** [comment]
@livarcocc that might be worth a shot.. can't you just cherry-pick your commit which fixed these tests from rc3 to this branch?

**Comment by rohit21agrawal** [comment]
@livarcocc @piotrpMSFT ... if we don't merge this in soon, we'll again have VS builds which load different versions of dependent DLLs and run into bunch of issues.. can someone help quickly?

CC: @rrelyea

**Comment by livarcocc** [comment]
Looking now.

**Comment by rohit21agrawal** [comment]
@piotrpMSFT approval?

File: TestAssets/TestPackages/dotnet-desktop-and-portable/dotnet-desktop-and-portable.csproj [hunk 1]
```diff
@@ -19,6 +19,10 @@
     </PackageReference>
   </ItemGroup>
 
+  <ItemGroup>
+    <PackageReference Include="NewtonSoft.Json" Version="9.0.1" />
```
Comment (piotrpMSFT) [inline TestAssets/TestPackages/dotnet-desktop-and-portable/dotnet-desktop-and-portable.csproj]: Please add a comment pointing to the issue that tracks whatever is happening here. @livarcocc mentioned that there is a bug that is fixed by this, but let's make sure it's tracked.

File: TestAssets/TestProjects/DependencyContextFromTool/DependencyContextFromTool.csproj [hunk 2]
```diff
@@ -18,6 +18,9 @@
       <Version>1.0.0-*</Version>
     </DotNetCliToolReference>
   </ItemGroup>
+  <ItemGroup>
+    <PackageReference Include="NewtonSoft.Json" Version="9.0.1" />
```
Comment (piotrpMSFT) [inline TestAssets/TestProjects/DependencyContextFromTool/DependencyContextFromTool.csproj]: Same here

**Comment by rohit21agrawal** [comment]
@dotnet-bot test Windows_NT x64 Release Build please

**Pull request merged**